FirebaseRemoteConfigSwift.podspec

```diff
@@ -38,6 +38,7 @@ app update.
     'FirebaseRemoteConfigSwift/Sources/**/*.swift',
   ]
 
+  s.dependency 'FirebaseCore', '~> 10.0'
```

Comment:
The test app surfaced an RC issue where RCSwift uses a FirebaseApp API, yet doesn't directly depend on FirebaseCore. I'm unable to reproduce this in a fresh project locally, but explicitly adding the `FirebaseCore` dependency fixes the issue.

FWIW, while this lone will keep CI green on master, it will likely not be released as I will be able to remove it in #11808.